modules/Evse15118D20/charger/ISO15118_chargerImpl.cpp

```diff
@@ -213,6 +286,85 @@ iso15118::session::feedback::Callbacks ISO15118_chargerImpl::create_callbacks()
         publish_dc_ev_target_voltage_current({target_voltage, 0});
     };
 
+    callbacks.notify_ev_charging_needs =
+        [this](const dt::ServiceCategory& service_category, const dt::AcConnector& ac_connector,
+               const dt::ControlMode& control_mode, const dt::MobilityNeedsMode& mobility_needs_mode,
+               const feedback::EVSE_TransferLimits& evse_limits, const feedback::EV_TransferLimits& ev_limits,
+               const feedback::EV_SEControlMode& ev_control_mode) {
+            // Everest types sent to OCPP
+            using namespace types::iso15118;
+
+            ChargingNeeds charging_needs;
+
+            EnergyTransferMode requested_energy_transfer = EnergyTransferMode::AC_single_phase_core;
+            if (service_category == dt::ServiceCategory::AC) {
+                if (ac_connector == dt::AcConnector::SinglePhase) {
+                    requested_energy_transfer = EnergyTransferMode::AC_single_phase_core;
+                } else if (ac_connector == dt::AcConnector::ThreePhase) {
+                    requested_energy_transfer = EnergyTransferMode::AC_three_phase_core;
+                }
+            } else if (service_category == dt::ServiceCategory::AC_BPT) {
+                requested_energy_transfer = EnergyTransferMode::AC_BPT;
+            } else if (service_category == dt::ServiceCategory::DC) {
+                requested_energy_transfer = EnergyTransferMode::DC;
+            } else if (service_category == dt::ServiceCategory::DC_ACDP) {
+                requested_energy_transfer = EnergyTransferMode::DC_ACDP;
+            } else if (service_category == dt::ServiceCategory::DC_BPT) {
+                requested_energy_transfer = EnergyTransferMode::DC_BPT;
+            } else if (service_category == dt::ServiceCategory::DC_ACDP_BPT) {
+                requested_energy_transfer = EnergyTransferMode::DC_ACDP_BPT;
+            }
```

Comment:
Move this to a local function. E.g. `EnergyTransferMode get_energy_transfer_mode(dt::ServiceCategory, dt::AcConnector)`